lightning/lib/bolt12/src/lib.rs

```diff
@@ -0,0 +1,107 @@
+extern crate lightning;
+extern crate bech32;
+
+use std::convert::TryFrom;
+use std::ffi::{c_char, CStr};
+use lightning::offers::offer::Amount;
+use bech32::FromBase32;
+use lightning::bitcoin::secp256k1::PublicKey;
+use lightning::offers::invoice::Bolt12Invoice;
+
+const BECH32_BOLT12_INVOICE_HRP: &str = "lni";
+
+fn parse_offer(offer: *const c_char) -> Result<lightning::offers::offer::Offer, String> {
+    let raw = unsafe { CStr::from_ptr(offer) };
+    match raw.to_str() {
+        Ok(res) => res.parse::<lightning::offers::offer::Offer>().map_err(|_| "Failed to parse offer".to_string()),
```

Comment:
Why are you ignoring the error that ldk throws? Also, I'd prefer those errors in here to start with a lowercase letter